Update Dependencies

drrtdrrt opened the pull request:
This PR updates some of the python and rust dependencies to address security advisories. While we are here, we will update the dependencies in rust across the board as many were quite outdated.

drrtdrrt committed:
update python dependencies

drrtdrrt committed:
update rust dependencies

drrtdrrt committed:
update rust dependencies and bump a few minor versions

drrtdrrt committed:
set serde to be a workspace version for all crates

drrtdrrt committed:
another round of dependency version bumps in rust

drrtdrrt committed:
update testbed server dependencies

drrtdrrt committed:
bump a few more dependencies

drrtdrrt committed:
update code to work with new axum version

JollyJellyBerry reviewed (approved):
Long and tedious job. It all looks good to me. Did you verify if we were using all the dependencies we were installing as part of the install?

drrtdrrt committed:
merge develop into update-deps

drrtdrrt commented:
> Long and tedious job. It all looks good to me. Did you verify if we were using all the dependencies we were installing as part of the install?

Yeah double checked what was using what, and fixed some duplication of dependencies that were used across the workspace and added them to the parent `Cargo.toml`. Cargo seems to make this quite easy as opposed to python/poetry.
